### PR TITLE
[BPK-1374] Add `accessoryView` for native input

### PR DESCRIPTION
**Pull request by k0nserv**
Adds the new `accessoryView`  to our React Native text inputs.

## Todo

+ [x] Fix RTL for descriptions
+ [x] Changelog
+ [x] Docs update

## Visuals

screenshot_1519837534
simulator screen shot - iphone 6 - 2018-02-28 at 17 05 29

### CODE REVIEW

File: native/packages/react-native-bpk-component-text-input/src/BpkTextInput.js [hunk 1]
```diff
@@ -146,8 +155,11 @@ class BpkTextInput extends Component<Props, State> {
       valid,
       onFocus,
       onBlur,
+      accessoryView,
       ...rest
     } = this.props;
+    const hasAccessoryView = accessoryView != null;
```
Comment (matthewdavidson) [inline native/packages/react-native-bpk-component-text-input/src/BpkTextInput.js]: should use `!==`

Comment (k0nserv) [reply native/packages/react-native-bpk-component-text-input/src/BpkTextInput.js]: Yeah I think I had a thought here, but it's probably a dumb idea anyway

**Comment by k0nserv** [comment]
RTL is actually fine if you use a LTR string in Android and an LTR bundle in iOS. The container is full width.

**Comment by matthewdavidson** [comment]
Small bug with label colour in inputs with value vs no value:
screen shot 2018-03-01 at 13 46 17

Both examples should have the dark colour I would have thought?

File: native/packages/react-native-bpk-component-text-input/stories.js [hunk 2]
```diff
@@ -62,57 +72,69 @@ class StatefulBpkTextInput extends Component<
...
+  .add('docs:text-inputs-with-accessory-view', () => (
+    <ScrollView>
+      <StatefulBpkTextInput
+        label="Phone number"
+        initialValue="+441234567890"
+        keyboardType="phone-pad"
+        style={styles.input}
+        accessoryView={<View style={styles.accessoryView} />}
+      />
+    </ScrollView>
```
Comment (matthewdavidson) [inline native/packages/react-native-bpk-component-text-input/stories.js]: Can we add an example with a placeholder and no initial value?

e.g.
```
.add('docs:text-inputs-with-accessory-view', () => (
    <ScrollView>
      <StatefulBpkTextInput
        label="Phone number"
        initialValue="1234567890"
        keyboardType="phone-pad"
        style={styles.input}
        accessoryView={<View style={styles.accessoryView} />}
      />
      <StatefulBpkTextInput
        label="Phone number"
        initialValue=""
        keyboardType="phone-pad"
        style={styles.input}
        placeholder="E.g. 1234567890"
        accessoryView={<View style={styles.accessoryView} />}
      />
    </ScrollView>
  ));
```

Comment (k0nserv) [reply native/packages/react-native-bpk-component-text-input/stories.js]: Yeah sure

**Comment by k0nserv** [comment]
> Both examples should have the dark colour I would have thought?

Yes you are right

File: native/packages/react-native-bpk-component-text-input/stories.js (hunk 2 above)
Comment (matthewdavidson) [reply native/packages/react-native-bpk-component-text-input/stories.js]: Actually can we not add all the other examples too? error, success etc?

**Pull request merged**